pkg/controller/pd_control.go

```diff
@@ -32,6 +32,34 @@ func GetPDClient(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) p
 	return pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.IsTLSClusterEnabled())
 }
 
+// GetPDClient tries to return an available PDClient
+// If the pdClient built from the PD service name is unavailable, try to
+// build another one with the ClientURL in the PeerMembers.
+// ClientURL example:
+// ClientURL: https://cluster2-pd-0.cluster2-pd-peer.pingcap.svc.cluster2.local
+func GetPDClient(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
+	pdClient := GetPDClientBasic(pdControl, tc)
+
+	if len(tc.Status.PD.PeerMembers) == 0 {
+		return pdClient
+	}
+
+	_, err := pdClient.GetHealth()
```

Comment:
Should we need to check the returned `HealthInfo` here?

Reply:
It just checks the connectivity here. If the in-cluster PD endpoint works, there's no need to switch to the PD in other clusters or somewhere else.

Reply:
If all of the in-cluster PDs are unhealthy, it still not switches to the PD in other clusters?

Reply:
The HTTP request will fail, the PD svc in this cluster will not respond to the requests. Then this function will return an error, we begin to find the available PD client in the PeerMembers. The connectivity is important here. If GetHealth works, return an "Unhealthy" result, the failover loop in tidb-operator, and the health check of the service in Kubernetes can handle these issues.